cmake/templates/pkgConfig.cmake.in

```diff
@@ -116,18 +137,31 @@ foreach(library ${libraries})
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
   elseif(TARGET ${library})
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
+  elseif(IS_ABSOLUTE ${library} AND CMAKE_CROSSCOMPILING)
+    get_filename_component(lib_path ${library} DIRECTORY)
+    get_filename_component(lib_name ${library} NAME)
+    set(lib "lib-NOTFOUND")
+    # use find_library to locate library given by absolute path nested under potential root path
+    find_library(lib ${lib_name}
+      PATHS ${lib_path}
+      NO_DEFAULT_PATH ${_find_root_path_both})
+    if(NOT lib)
+      message(FATAL_ERROR "Project '${PROJECT_NAME}' tried to find library '${library}'.  The library is neither a target nor built/installed properly.  Did you compile project '@PROJECT_NAME@'?  Did you find_package() it before the subdirectory containing its code is included?")
+    endif()
+    _list_append_unique(@PROJECT_NAME@_LIBRARY_DIRS ${lib_path})
+    list(APPEND @PROJECT_NAME@_LIBRARIES ${lib})
   elseif(IS_ABSOLUTE ${library})
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
   else()
     set(lib_path "")
-    set(lib "${library}-NOTFOUND")
+    set(lib "lib-NOTFOUND")
     # since the path where the library is found is returned we have to iterate over the paths manually
     foreach(path @PKG_CONFIG_LIB_PATHS@)
       find_library(lib ${library}
         PATHS ${path}
-        NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
+        NO_DEFAULT_PATH ${_find_root_path_both})
```

Comment:
Yes, the change of the argument changes the location but only if `CMAKE_FIND_ROOT_PATH` is set, which is afaik only the case in cross-compiled builds. In a regular build there will be no change.

The current behavior is only wrong when it comes to cross-compiled builds. In such builds, you usualy have two filesystems available, the host-machine's and the target's (under a prefix in the host filesystem). The person who builds can specify whether the libraries/files required for the build should be looked up in the host- or target-filesystem), or both. Usually you'd want to look up libraries in the rootfs since they need to have the correct architecture.

The current behavior overrides this and forces the lookup to be only on the host-filesystem. This will break for every build since the library will not be found (if the host does not have the library installed to the same location) or fail when compiling (if installed to the same location) since the architectures won't match.

This is even unrelated from the introduced flag, which is required when not installing but rather building to devel-space where we want to find a target-architecture-library with a host-path.